src/Voron/Data/Containers/Container.cs

```diff
@@ -499,12 +501,39 @@ void RemovePageFromContainerFreeList(Container parent, Container page)
             }
 
             void AddPageToContainerFreeList(Container parent, Container page)
+            {
+                page.Header.OnFreeList = true;
+                ModifyMetadataList(llt, parent, ContainerPageHeader.FreeListOffset, add: true, page.Header.PageNumber);
+            }
+            
+            void AddPageToContainerAllPagesList(Container parent, Container page)
             {
                 page.Header.OnFreeList = true;
                 ModifyMetadataList(llt, parent, ContainerPageHeader.AllPagesOffset, add: true, page.Header.PageNumber);
             }
         }
 
+        private static IEnumerable<long> ScanFreePages(Tree freeList)
+        {
```

Comment:
If we are in a high traffic section, I would rather do this without `IEnumerable<long>` and use a `struct` iterator.